transport-classes-io_uring/src/main/java/io/netty5/incubator/channel/uring/IOUringServerSocketChannel.java

```diff
@@ -0,0 +1,238 @@
+        readNow();
+    }
+
+    @Override
+    protected boolean doReadNow(ReadSink readSink) throws IOException {
+        int res = currentCompletionResult;
+        short data = currentCompletionData;
+        if (data != IS_ACCEPT) {
+            readSink.processRead(0, 0, null);
+            return false;
+        }
+        currentCompletionResult = 0;
+        currentCompletionData = 0;
+        if (res >= 0) {
+            Channel channel = newChildChannel(res);
+            readSink.processRead(1, 1, channel);
+        } else if (res == ERRNO_EAGAIN_NEGATIVE || res == ERRNO_EWOULDBLOCK_NEGATIVE) {
+            // Check if we failed because there was nothing to accept atm.
+            readSink.processRead(0, 0, null);
+        } else {
+            // Something bad happened. Convert to an exception.
+            throw Errors.newIOException("io_uring accept", res);
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean processRead(ReadSink readSink, Object read) {
+        throw new UnsupportedOperationException();
+    }
+
+    private Channel newChildChannel(int fd) {
+        final SocketAddress peer;
+        if (socket.protocolFamily() == SocketProtocolFamily.UNIX) {
+            peer = null;
+        } else {
+            peer = buildAddress();
+        }
+        return new IOUringSocketChannel(
+                this, childEventLoopGroup().next(), false,
+                new AdaptiveReadHandleFactory(), new SocketChannelWriteHandleFactory(Integer.MAX_VALUE, SSIZE_MAX),
+                LinuxSocket.wrapBlocking(fd, socket.protocolFamily()), peer, true);
+    }
+
+    private SocketAddress buildAddress() {
+        if (socket.isIpv6()) {
+            return SockaddrIn.readIPv6(sockaddrPtr, inet6AddressArray, inet4AddressArray);
+        }
+        return SockaddrIn.readIPv4(sockaddrPtr, inet4AddressArray);
+    }
+
+    @Override
+    protected boolean doConnect(SocketAddress remoteAddress, SocketAddress localAddress, Buffer initialData) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    protected Object filterOutboundMessage(Object msg) throws Exception {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    protected void submitAllWriteMessages(WriteSink writeSink) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    void writeComplete(int result, long udata) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    protected void doShutdown(ChannelShutdownDirection direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean isShutdown(ChannelShutdownDirection direction) {
+        return !isActive();
+    }
+
+    @Override
+    protected InternalLogger logger() {
+        return LOGGER;
+    }
+
+    @Override
+    protected void doClose() {
+        super.doClose();
+        free(sockaddrMemory);
```

Comment:
I think we want to Tod all of this in a finally block to ensure nothing leaks in all cases.